src/jit/codegencommon.cpp

```diff
@@ -7557,6 +7557,7 @@ void CodeGen::genPrologPadForReJit()
 regMaskTP           CodeGen::genPInvokeMethodProlog(regMaskTP initRegs)
 {
     assert(compiler->compGeneratingProlog);
+    noway_assert((compiler->opts.eeFlags & CORJIT_FLG_PINVOKE_USE_HELPERS) == 0);
```

Comment:
Rather than duplicating this check at multiple places, i would suggest to abstract it in a routine in Compiler::Options struct as  shouldUsePInvokeHelpers()

e.g. if (compiler->opts.shouldUsePInvokeHelpers())